sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -169,6 +170,7 @@ class Engine(val config: EngineConfig) {
         disclosedContractIds,
         disclosedKeyHashes,
       )
+      _ <- preprocessor.getInputCost
```

Comment:
getInputCost has a `Result[?]` return type, so it can immediately halt the submission flow is input sizes are too large